components/VoteCommentModal.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { FunctionComponent, useState } from 'react'
+import { postChatMessage } from '../actions/chat/postMessage'
+import { ChatMessageBody, ChatMessageBodyType } from '../models/chat/accounts'
+import { RpcContext } from '../models/core/api'
+import useWalletStore from '../stores/useWalletStore'
+import useRealm from '../hooks/useRealm'
+import { castVote } from '../actions/castVote'
+import { Vote } from '../models/instructions'
+import Button, { LinkButton } from './Button'
+// import { notify } from '../utils/notifications'
+import Loading from './Loading'
+import Modal from './Modal'
+import Input from './Input'
+import Tooltip from './Tooltip'
+
+interface VoteCommentModalProps {
+  onClose: () => void
+  isOpen: boolean
+  vote: Vote
+}
+
+const VoteCommentModal: FunctionComponent<VoteCommentModalProps> = ({
+  onClose,
+  isOpen,
+  vote,
+}) => {
+  const [submitting, setSubmitting] = useState(false)
+  const [comment, setComment] = useState('')
+  const wallet = useWalletStore((s) => s.current)
+  const connection = useWalletStore((s) => s.connection)
+  const { proposal } = useWalletStore((s) => s.selectedProposal)
+  const { fetchChatMessages } = useWalletStore((s) => s.actions)
+  const { fetchVoteRecords } = useWalletStore((s) => s.actions)
+  const { realm, ownTokenRecord } = useRealm()
+
+  const submitVote = async (vote: Vote) => {
+    setSubmitting(true)
+    const rpcContext = new RpcContext(
+      proposal.account.owner,
+      wallet,
+      connection.current,
+      connection.endpoint
+    )
+
+    const msg = new ChatMessageBody({
+      type: ChatMessageBodyType.Text,
+      value: comment,
+    })
+
+    try {
+      await castVote(
+        rpcContext,
+        realm.pubkey,
+        proposal,
+        ownTokenRecord.pubkey,
+        vote
+      )
+      if (comment) {
```

Comment:
It would be better UX to combine vote and chat instruction into a single transaction to avoid two wallet confirmations